### PR TITLE
 taito_zm: DSP emulation (work in progress!)

### PR DESCRIPTION
**Pull request by superctr**
Still working on this, but I'm opening the PR in order to bring some attention...

This PR improves sound emulation in Taito Zoom sound system games. This includes FX-1B and GNet boards. Comes at a fairly big performance hit, but I think you'd rather take that than the really tinny sound from before?

### CODE REVIEW

**Comment by superctr** [comment]
I'll look into the issue regarding ``psyvarrv`` when I get home. All GNet games use the same DSP code, so it's possible it's related to the raycrisis song 9 issue.

**Comment by superctr** [comment]
I was still working on this...

**Comment by rb6502** [comment]
Please don't open PRs you don't want merged then :)

**Comment by superctr** [comment]
How else should I get feedback then? Like that Psyvariar issue, I might not have found it otherwise. Anyway, I looked at that, it's weird, the CPU a set of values of values to the DSP which looks like a value and what is supposed to be the same value, but inverted. Except the sign bit actually isn't set, making it a huge positive integer instead.

**Comment by balr0g** [comment]
> How else should I get feedback then?

The best way is by putting **DO NOT MERGE** in the PR text.

**Comment by superctr** [comment]
Ok, I thought I was clear with the "WIP" tag, but I'll just do that next time then.

**Comment by rb6502** [comment]
The best way to get operational feedback is to have it merged so everyone can test it.  Non-merge PRs in the future will be closed summarily.

**Comment by MooglyGuy** [comment]
And summarily reopened, because there's nothing wrong with submitting a PR in order to get feedback before it goes in. That's how we've operated for months now, and how we will continue to operate. If you don't feel like giving feedback, don't give it, but let others do what they wish.

**Comment by rb6502** [comment]
We have *never* operated that way.  You get testing by having your changes go in like a big boy and letting people bang on them.

**Comment by rb6502** [comment]
The only kind of submit-for-feedback we have supported is on coding style/standards.

**Comment by MoochMcGee** [comment]
Oh come on, Arbee, MOST projects prefer if you submit a PR in the meantime, just so they can review your approach and shit, and so the coder doesn't end up wasting nearly as much of their time.

File: src/mame/audio/taito_zm.h [hunk 1]
```diff
@@ -14,12 +14,15 @@
 #include "cpu/tms57002/tms57002.h"
 #include "sound/zsg2.h"
 
+#define USE_DSP               // Uncomment when DSP emulation is working
```
Comment (cuavas) [inline src/mame/audio/taito_zm.h]: This name is far too generic for something in a header that needs to be included in downstream driver sources.  Could someone please either remove it completely if the DSP is in fact working, or at least prefix it with something to make it less likely to class with something equally generically named in another header?

**Pull request merged**